src/diagram/models/variable.ts

```diff
@@ -351,5 +348,13 @@ export const Variable = types.model("Variable", {
   setExpression(newExpression?: string) {
     self.expression = newExpression;
   },
+}))
+.actions(self => ({
+  removeInput(input: DQNodeType) {
```

Comment:
The variable model shouldn't have any dependency on the DQNode. 
So this should be changed to take either a variable instance instead.